### DISCUSSION TITLE
Retrieving values from DEM data

### DISCUSSION THREAD

**Discussion by harrisonjkaplan** [google/Xee#134] [Q&A] [answered]
Hi there, 
I am relatively new to GEE and am trying to get DEM raster data within a certain geometry. 

```
dem = ee.Image('USGS/3DEP/10m').select('elevation')
dem_ic = ee.ImageCollection(dem)
rect = ee.Geometry.Rectangle(-125.0, 43.0, -125.1, 43.1)

ds = xr.open_dataset(
    dem_ic,
    engine='ee',
    geometry=rect,
)

da = ds['elevation'] 
```
I am getting error "No numeric data to plot." When I run da.plot(), and the shape of `da.values` is (1, 1, 1)

I am wondering if I am gathering the data incorrectly.

Any help would be appreciated, 

Thanks!

**Comment by jdbcode** [comment, marked as answer]
There are two things going wrong here:

1. Your region is outside of the dataset. Your rectangle is in the Pacific ocean, off the coast of Oregon.

![image](https://github.com/user-attachments/assets/f1053845-f562-43d5-96b8-dff9c54d428f)

2. You are accepting the default `crs` and `scale`, which are 'EPSG:4326' and 1 degree, respectively. And, with the small rectangle you've defined, it only interests a single 1-degree pixel. So, you only get 1 value returned, which can't be plotted with `.plot()`.

The solution in your case is to move the rectangle to a region that has data and to provide the `crs` and `scale`  parameters based on the dataset. For example:

```py
dem = ee.Image('USGS/3DEP/10m').select('elevation')
dem_ic = ee.ImageCollection(dem)
rect = ee.Geometry.Rectangle(-124.0302, 42.8196, -123.9258, 42.9243)

proj = dem_ic.first().select(0).projection().getInfo()

ds = xr.open_dataset(
    dem_ic,
    engine='ee',
    geometry=rect,
    crs=proj['crs'],
    scale=proj['transform'][0],
)

da = ds['elevation'] 

display(da)
da.plot()
```

![image](https://github.com/user-attachments/assets/2c819eb1-b717-40b0-9b54-795520296e94)
